docs/01-on-chain-passport-attestation.md

```diff
@@ -0,0 +1,74 @@
+This will allow capturing a snapshot of a users passport on-chain.
+
+```js
+new SchemaEncoder(
+   "bytes32[] providers, bytes32[] hashes, uint64[] issuanceDates"
+);
+```
+
+**providers**
+- this field shall indicate which stamps (i. e. which providers) a user has in his passport. This field is declared as an array of `bytes32` but it shall be used as a array of bits, meaning:
+- we will use an ordered list of providers, and we will assign each of the providers a position in the bytes32 array and a bit, for example: 
+    - position 0, bit 0 (`0x0000000000000001`): Brightid
+    - position 0, bit 1 (`0x0000000000000002`): unityStakingBronze
+    - position 0, bit 2 (`0x0000000000000004`): CommunityStakingGold
+    - position 0, bit 3 (`0x0000000000000008`): CommunityStakingSilver
+    - position 0, bit 4 (`0x000000000000000f`): Coinbase
+    - position 0, bit 5 (`0x0000000000000010`): Discord
+- we will set the coresponding bit of the element in the `providers` field to 0, if a user does have the stamp for the providers who is assigned that bit
+
+**hashes**
+- this field shall record the individual hashes for each stamp a user owns:
+- this will be an ordered list, the hashes will be writted in the order of the providers as defined for the `providers` field
+- this field will only record the hashes for the stamps that a user owns, meaning we will skip any elements where the provider bit is set to 0 in the `providers` field
+
+**issuanceDates**
+- (unix timestamp) individual issuance dates for the stamps. Similar to the hashes field, this is a ordered array. The issuance date can be different form the EAS creation timestamp.
+
+
+Considering the list of providers abovem if a user has the `BrightId`, `CommunityStakingSilver` and `Discord` stamps, his attestation will look like:
+```json
+{
+    "providers": ["0x0000000000000019"],  
+    "hashes": ["0x0000000000000001", "0x0000000000000002", "0x0000000000000003"],  
+    "issuanceDates": ["123456789", "123456789", "123456789"],  
+}
+```
+The 3 bits coresponding to each position of the providers will be set to 1 in the first element in providers, and we record the 3 hashes in the `hashes` field.
+
+# GitcoinPassportResolver
+For our use-case it is important that given an ETH address of a user, we are able to determine all the stamps that the user owns.
+The EAS smart contract does not offer this functionality.
+However we can create a resolver smart contract that stores the ***latest*** passport attestation for a particular recipient:
+
+    recipient => attestation UUID
+
+In solidity this would translate to:
+
+    mapping(address => bytes32) public passports;
+
+Given this structure, we would be able to retrieve the UUID for the latest passport attestation of a particular recipient.
+
+As also mentioned in [On Chain Data](./00-on-chain-data.md), the resolver smart contract will:
+- only allow calls from a trusted EAS smart contract
+- data coming from a truster Attester
```

Comment:
Is this supposed to say "only accept data coming from a trusted attester"?

Reply:
Correct. Have fixed that now.